### DISCUSSION TITLE
Does not work with `favicons-webpack-plugin`

### DISCUSSION THREAD

**Discussion by ChocolateLoverRaj** [vercel/next.js#29223] [Help] [answered]
### What version of Next.js are you using?

11.1.2

### What version of Node.js are you using?

14.16.0

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying your application?

Vercel

### Describe the Bug

I am trying to use [favicons-webpack-plugin](https://github.com/jantimon/favicons-webpack-plugin).

This is my `next.config.js`:
```js
const FaviconsWebpackPlugin = require("favicons-webpack-plugin");

module.exports = {
  webpack: (config) => {
    (config.plugins ?? (config.plugins = [])).push(
      new FaviconsWebpackPlugin({
        logo: "./icon.png",
        outputPath: "./public",
        inject: true
      })
    );
    return config;
  }
};
```

No favicons are injected and going to `/favicon.png` has `404`. It doesn't work in `dev` or `start`.

### Expected Behavior

The favicons should be served like a `public` path and should be shown in page.

### To Reproduce

Code sandbox: https://codesandbox.io/s/next-js-with-favicons-webpack-plugin-7ye1d?file=/next.config.js:0-320.

**Comment by ijjk** [comment]
Hi, Next.js doesn't process HTML files through webpack by default so this plugin isn't expected to work like this. You can add favicons to your document using a custom document, see more info here https://nextjs.org/docs/advanced-features/custom-document

**Reply by ChocolateLoverRaj** [reply]
The link does show how it's possible to add favicons to a Next.js project. However, I like `favicons-webpack-plugin` because it automatically does all the favicon stuff for me. Because Next.js isn't as flexible as webpack, there is no clear solution for icons. People shouldn't have to spend hours adjusting to a Next.js specific thing, or have to make custom plugins like `next-`some feature for it to work with Next.js.

**Comment by agusterodin** [comment, marked as answer]
I wish they had it out of the box (or some sort of officially sanctioned solution), something similar to gatsby-plugin-manifest. In the meantime here are two solutions:

https://github.com/agusterodin/Modern-Favicons (made by me, very lightweight)
https://github.com/tinialabs/next-favicon-loader (more of a gatsby-plugin-manifest equivalent)

**Comment by ChocolateLoverRaj** [comment]
Sometimes the following solution also works: 
Switch from Next.js to just configuring webpack manually, if you're fine with not having SSR. Then `favicons-webpack-plugin` will work like their zero config example.
